compiler/rustc_hir_analysis/src/collect/type_of.rs

```diff
@@ -947,7 +947,7 @@ fn infer_placeholder_type<'a>(
 
     // Typeck doesn't expect erased regions to be returned from `type_of`.
     tcx.fold_regions(ty, |r, _| match *r {
-        ty::ReErased => tcx.lifetimes.re_static,
+        ty::ReErased | ty::ReError => tcx.lifetimes.re_static,
```

Comment:
why don't we keep the ReError around here?

Reply:
Neither change had any effect in the test suite. Removing.